core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -205,6 +216,19 @@ public static RequestBody fromContentProvider(ContentStreamProvider provider, lo
         return new RequestBody(provider, contentLength, mimeType);
     }
 
+    /**
+     * Creates a {@link RequestBody} from the given {@link ContentStreamProvider}.
+     *
+     * @param provider The content provider.
+     * @param contentLength The content length, using non primitive data type.
+     * @param mimeType The MIME type of the content.
+     *
+     * @return The created {@code RequestBody}.
+     */
+    public static RequestBody fromContentProvider(ContentStreamProvider provider, Long contentLength, String mimeType) {
```

Comment:
Should we just omit the contentLength param?

Reply:
Same as above, this will be a breaking change. But I can do the change if that matches with our standard better. I agree omitting this param can be a better customer experience.

Reply:
Why would adding `RequestBody fromContentProvider(ContentStreamProvider provider, String mimeType)` be a breaking change?

Reply:
Oh I was thinking about removing the `contentLength` param in the original `fromContentProvider` method, because the getter of `contentLength` is going to be removed, and the only source of `contentLength` would be the length of files or byte arrays, so I thought it's doable to remove the `contentLength` parameter from the origin `fromContentProvider` method, it's just a breaking change. But anyway, adding a new method should be totally fine.